packages/create-next-app/index.ts

```diff
@@ -44,13 +44,16 @@ const program = new Commander.Command(packageJson.name)
 
 async function run(): Promise<void> {
   // NodeJs Version Validation
-  var currentNodeVersion: Number = parseFloat(process.versions.node)
+  const currentNodeVersion: number = parseFloat(process.versions.node)
+  const supportedNodeVersion: number = parseFloat(
+    (packageJson?.engines.node).slice(2)
```

Comment:
[`semver`](https://www.npmjs.com/package/semver) will help you parse and compare versions better.